client/rest-high-level/src/main/java/org/elasticsearch/client/cluster/ClusterClient.java

```diff
@@ -41,7 +44,7 @@
 public final class ClusterClient {
     private final RestHighLevelClient restHighLevelClient;
 
-    ClusterClient(RestHighLevelClient restHighLevelClient) {
+    public ClusterClient(RestHighLevelClient restHighLevelClient) {
```

Comment:
I worry that we open up the possibility for manually creating the sub-clients, potentially even providing a subclass of our main client. I am paranoid I know, but ideally we'd control this. After all there is no reason to open this up to users, it's another side-effect of the package/visibility changes.